modules/reactor/src/meshgenerators/CoreMeshGenerator.C

```diff
@@ -410,55 +416,78 @@ CoreMeshGenerator::CoreMeshGenerator(const InputParameters & parameters)
+        {
+          // For assembly structures, store region ID and block names of assembly regions and
+          // constituent pins
+          std::map<subdomain_id_type, std::vector<std::vector<subdomain_id_type>>>
+              pin_region_id_map = getMeshProperty<
+                  std::map<subdomain_id_type, std::vector<std::vector<subdomain_id_type>>>>(
+                  RGMB::pin_region_id_map, assembly);
+          for (auto pin = pin_region_id_map.begin(); pin != pin_region_id_map.end(); ++pin)
+            if (_pin_region_id_map.find(pin->first) == _pin_region_id_map.end())
+              _pin_region_id_map.insert(
+                  std::pair<subdomain_id_type, std::vector<std::vector<subdomain_id_type>>>(
+                      pin->first, pin->second));
+
+          std::map<subdomain_id_type, std::vector<std::vector<std::string>>> pin_block_name_map =
+              getMeshProperty<std::map<subdomain_id_type, std::vector<std::vector<std::string>>>>(
+                  RGMB::pin_block_name_map, assembly);
+          for (auto pin = pin_block_name_map.begin(); pin != pin_block_name_map.end(); ++pin)
+            if (_pin_block_name_map.find(pin->first) == _pin_block_name_map.end())
+              _pin_block_name_map.insert(
+                  std::pair<subdomain_id_type, std::vector<std::vector<std::string>>>(pin->first,
+                                                                                      pin->second));
+
+          // Define background and duct region ID map from constituent assemblies
+          if (_background_region_id_map.find(assembly_type) == _background_region_id_map.end())
+          {
+            // Store region ids and block names associated with duct and background regions for each
+            // assembly, in case block names need to be recovered from region ids after
+            // multiple assemblies have been stitched together into a core
+            std::vector<subdomain_id_type> background_region_ids =
+                getMeshProperty<std::vector<subdomain_id_type>>(RGMB::background_region_id,
+                                                                assembly);
+            std::vector<std::vector<subdomain_id_type>> duct_region_ids =
+                getMeshProperty<std::vector<std::vector<subdomain_id_type>>>(RGMB::duct_region_ids,
+                                                                             assembly);
+            _background_region_id_map.insert(
+                std::pair<subdomain_id_type, std::vector<subdomain_id_type>>(
+                    assembly_type, background_region_ids));
+            _duct_region_id_map.insert(
+                std::pair<subdomain_id_type, std::vector<std::vector<subdomain_id_type>>>(
+                    assembly_type, duct_region_ids));
+
+            std::vector<std::string> background_block_names =
+                getMeshProperty<std::vector<std::string>>(RGMB::background_block_name, assembly);
```

Comment:
```suggestion
                getMeshProperty<std::vector<SubdomainName>>(RGMB::background_block_name, assembly);
```

Reply:
For same reason, unable to update data type since this variable is part of metadata store